Bump OVN-K

skitt opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

skitt committed:
Bump OVN-K

Signed-off-by: Stephen Kitt <skitt@redhat.com>

skitt commented:
It seems the obvious code change is incorrect.

yboaron commented:
QQ: do we need to update also [ovn-k version used on Kind](https://github.com/submariner-io/shipyard/blob/devel/scripts/shared/lib/clusters_kind#L277) to be in-sync with ovn-kubernetes/go-controller ?

skitt commented:
> QQ: do we need to update also [ovn-k version used on Kind](https://github.com/submariner-io/shipyard/blob/devel/scripts/shared/lib/clusters_kind#L277) to be in-sync with ovn-kubernetes/go-controller ?

I don’t think so, but it wouldn’t hurt — this was done in relation to https://github.com/submariner-io/submariner-website/issues/1223. In any case the unit test failure needs to be addressed; that’s independent of the image bump :-/.

yboaron commented:
> > QQ: do we need to update also [ovn-k version used on Kind](https://github.com/submariner-io/shipyard/blob/devel/scripts/shared/lib/clusters_kind#L277) to be in-sync with ovn-kubernetes/go-controller ?
> 
> I don’t think so, but it wouldn’t hurt — this was done in relation to [submariner-io/submariner-website#1223](https://github.com/submariner-io/submariner-website/issues/1223). In any case the unit test failure needs to be addressed; that’s independent of the image bump :-/.

Ooops ,  I missed https://github.com/submariner-io/shipyard/pull/1830#pullrequestreview-2578699965
Thanks

yboaron commented:
Probably [fake ovn ovsdb_client](https://github.com/submariner-io/submariner/blob/devel/pkg/routeagent_driver/handlers/ovn/fake/ovsdb_client.go) should also be updated

skitt commented:
https://github.com/ovn-kubernetes/ovn-kubernetes/pull/3544 is the PR that breaks Submariner, but I haven’t figured out why.

dfarrell07 commented:
It would be interesting to manually rebase this and see how CI is doing.

skitt commented:
```
Summarizing 2 Failures:
  [FAIL] Handler when a GatewayRoute is created and deleted [It] should correctly reconcile OVN router policies
  /home/skitt/go/src/github.com/submariner-io/submariner/pkg/routeagent_driver/handlers/ovn/fake/ovsdb_client.go:181
  [FAIL] Handler when NonGatewayRoutes are created, updated and deleted [It] should correctly reconcile OVN router policies
  /home/skitt/go/src/github.com/submariner-io/submariner/pkg/routeagent_driver/handlers/ovn/fake/ovsdb_client.go:181
```

skitt commented:
The route agent shows this error:

```
Event handler \"ovn-hostroutes-handler\" failed to initialize: error getting connection handler to connect to OvnDB: error creating NBDB connection: error connecting to ovsdb: failed to connect to tcp:ovnkube-db.ovn-kubernetes:6641: database OVN_Northbound validation error (9): database model contains a model for table Sample_Collector that does not exist in schema. Mapper Error. Object type nbdb.LogicalRouterPort contains field DhcpRelay (*string) ovs tag dhcp_relay: Column does not exist in schema. database model contains a model for table Sampling_App that does not exist in schema. Mapper Error. Object type nbdb.LogicalRouterPolicy contains field BFDSessions ([]string) ovs tag bfd_sessions: Column does not exist in schema. Mapper Error. Object type nbdb.DNS contains field Options (map[string]string) ovs tag options: Column does not exist in schema. Mapper Error. Object type nbdb.ACL contains field SampleEst (*string) ovs tag sample_est: Column does not exist in schema. database model contains a model for table Sample that does not exist in schema. Mapper Error. Object type nbdb.NAT contains field Match (string) ovs tag match: Column does not exist in schema. database model contains a model for table DHCP_Relay that does not exist in schema
```

Bumping OVN-K in Submariner probably requires bumping the reference commit and OVN image hash in Shipyard. To test that, Shipyard needs to be improved to allow external overrides of those two values, in the deployment script and in the `e2e` GHA.

dfarrell07 commented:
@skitt What's the status here? Still broken but we don't know why?

skitt commented:
> @skitt What's the status here? Still broken but we don't know why?

Exactly! Hopefully with the team ramping up on OVN-K you’ll be able to figure it out :wink:.

dfarrell07 commented:
CC @yboaron @aswinsuryan can you guys take a look at this one, see what the current status is?

dfarrell07 commented:
Closing in favor of https://github.com/submariner-io/submariner/pull/3728. -
